lib/core.ps1

```diff
@@ -142,11 +142,63 @@ function set_config {
         $scoopConfig.PSObject.Properties.Remove($name)
     }
 
+    # Initialize config's change
+    Complete-ConfigChange -Name $name -Value $value
+
     # Save config with UTF8NoBOM encoding
     ConvertTo-Json $scoopConfig | Out-UTF8File -FilePath $configFile
     return $scoopConfig
 }
 
+function Complete-ConfigChange {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory, Position = 0)]
+        [string]
+        $Name,
+        [Parameter(Mandatory, Position = 1)]
+        [AllowEmptyString()]
+        [string]
+        $Value
+    )
+
+    if ($Name -eq 'use_isolated_path') {
+        . "$PSScriptRoot\..\lib\system.ps1"
+
+        if ($Value -eq $true) {
+            info 'Switching to isolated path... This may take a while, please wait.'
+            $movedPath = Remove-Path -Path "$scoopdir\apps\*" -Quiet -PassThru
```

Comment:
`apps\scoop`?

Reply:
That dir is not in PATH, or I'm missing sth?

Reply:
`scoop\shims` is there, and we only shim `scoop.ps1`.